addons/web_editor/static/lib/odoo-editor/src/commands/commands.js

```diff
@@ -448,25 +450,51 @@ export const editorCommands = {
     },
     unlink: editor => {
         const sel = editor.document.getSelection();
-        // we need to remove the contentEditable isolation of links
-        // before we apply the unlink, otherwise the command is not performed
-        // because the content editable root is the link
-        const closestEl = closestElement(sel.focusNode, 'a');
-        if(closestEl) {
-            closestEl.removeAttribute('class'); // To prevent firefox from adding unnecessary <span>.
-        }
-        if (closestEl && closestEl.getAttribute('contenteditable') === 'true') {
-            editor._activateContenteditable();
+        const isCollapsed = !!sel.isCollapsed;
+        if (!isCollapsed) {
+            getDeepRange(editor.editable, { sel, splitText: true, select: true });
+            // Split leading and trailing links.
+            let { anchorNode, focusNode, anchorOffset, focusOffset } = sel;
+            const direction = getCursorDirection(anchorNode, anchorOffset, focusNode, focusOffset);
+            let startLink = closestElement(anchorNode, 'a');
+            if (startLink) {
+                startLink = splitAroundUntil(anchorNode, startLink);
+                const range = sel.getRangeAt(0);
+                anchorNode = startLink;
+                anchorOffset = direction === DIRECTIONS.RIGHT ? 0 : nodeSize(startLink);
+                range.setStart(anchorNode, anchorOffset);
+                sel.removeAllRanges();
+                sel.addRange(range);
+                sel.extend(focusNode, focusOffset);
+                getDeepRange(editor.editable, { sel, select: true });
+            }
+            let endLink = closestElement(focusNode, 'a');
+            if (endLink) {
+                endLink = splitAroundUntil(focusNode, endLink);
+                const range = sel.getRangeAt(0);
+                range.setStart(anchorNode, anchorOffset);
+                sel.removeAllRanges();
+                sel.addRange(range);
+                sel.extend(endLink, direction === DIRECTIONS.RIGHT ? nodeSize(endLink) : 0);
```

Comment:
Same here, obviously. Also, I'm confused because you seem to handle the case where there are two links, one at start and one at end, but what if there are multiple links inside the selection ? Like I'm selecting 4 different links ? The 2 links "inside" are going to be completely ignored by the lines above and therefore won't be split, right ? To be honest, I don't even understand why we need to split in the first place (and it is not explained anywhere) so I can't really say if that's a big deal or not.

Reply:
So as I explained above, this is meant to handle cases like `<a>a[b]c</a>` -> `<a>a</a>[b]<a>c</a>` so we don't need to split the middle links:

```js
<a>a[b</a>c<a>d</a>e<a>f]g</a>
```
->
```js
<a>a</a>[bcdef]<a>g</a>
```
There was no splitting to be done on that middle link since we're just unlinking it entirely.

Reply:
Oh, ok, waw that is _not_ obvious from the code... ^^' Do you think you could write some explanations in comments ? I'd even use the precise example that you used here in the message. It shows partial unlinking at the edges of the selection and complete unlinking in the middle, it's the perfect example to put in the comment imo.